### PR TITLE
add Notification Type to Snack Messaging

### PR DESCRIPTION
**Pull request by fabian-bizfactory**

### CODE REVIEW

**Comment by Kadrian** [comment]
@fabian-bizfactory please let me know once you've looked through the diff and you think this is ready for review

**Comment by fabian-bizfactory** [comment]
Sorry, about the PR pull. Found a potential issue after creating it

File: frontend/src/js/snack-message/SnackMessage.tsx [hunk 1]
```diff
@@ -41,11 +48,17 @@ const ClearZone = styled("div")`
...
-  const message = useSelector<StateT, string | null>(
-    (state) => state.snackMessage.message,
+  const { message, notificationType } = useSelector<StateT, SnackMessageStateT>(
+    (state) => state.snackMessage,
   );
   const dispatch = useDispatch();
-  const resetMessage = () => dispatch(setMessage({ message: null }));
+  const resetMessage = () =>
+    dispatch(
+      setMessage({
```
Comment (Kadrian) [inline frontend/src/js/snack-message/SnackMessage.tsx:57]: Let's use the `resetMessage` action instead here

File: frontend/src/js/snack-message/reducer.ts [hunk 2]
```diff
@@ -4,12 +4,19 @@ import { Action } from "../app/actions";
 
 import { resetMessage, setMessage } from "./actions";
 
+export enum SnackMessageTypeT {
```
Comment (Kadrian) [inline frontend/src/js/snack-message/reducer.ts:7]: Ending types with a`T` is something we're trying to slowly move away from, I'd suggest calling it `SnackMessageType`

File: frontend/src/js/snack-message/SnackMessage.tsx [hunk 3]
```diff
@@ -7,13 +7,20 @@ import { useClickOutside } from "../common/helpers/useClickOutside";
 import FaIcon from "../icon/FaIcon";
 
 import { setMessage } from "./actions";
+import { SnackMessageStateT, SnackMessageTypeT } from "./reducer";
 
-const Root = styled("div")`
+const colorLookupTable: { [key in SnackMessageTypeT]: string } = {
```
Comment (Kadrian) [inline frontend/src/js/snack-message/SnackMessage.tsx:12]: Let's avoid calling variables by their "data structure" ("lookup table") and instead describe what they contain. Also `Record` is useful here:
```suggestion
const bgColorByType: Record<SnackMessageType, string> = {
```

File: frontend/src/js/snack-message/actions.ts [hunk 4]
```diff
@@ -1,11 +1,14 @@
...
 
+import { SnackMessageTypeT } from "./reducer";
+
 export type SnackMessageActions = ActionType<
   typeof setMessage | typeof resetMessage
 >;
 
 export const setMessage = createAction("snack-message/SET")<{
   message: string | null;
+  notificationType: SnackMessageTypeT;
```
Comment (Kadrian) [inline frontend/src/js/snack-message/actions.ts:11]: how about just calling it `type`? 
```suggestion
  type: SnackMessageType;
```

File: frontend/src/js/snack-message/SnackMessage.tsx [hunk 5]
```diff
@@ -7,13 +7,20 @@ import { useClickOutside } from "../common/helpers/useClickOutside";
...
+  [SnackMessageTypeT.ERROR]: "rgba(0, 0, 0, 0.75)",
+  [SnackMessageTypeT.SUCCESS]: "rgba(12, 100, 39, 0.75)", // #0C6427
```
Comment (Kadrian) [inline frontend/src/js/snack-message/SnackMessage.tsx:14]: Have you tested what it looks like? I think with the green, it's actually too transparent, how about using `0.9`?

**Review by Kadrian — CHANGES_REQUESTED** [review]
Looks good overall, I've got a couple of suggestions, would be happy if you could take a look at those!

File: frontend/src/js/snack-message/SnackMessage.tsx (hunk 5 above)
Comment (fabian-bizfactory) [reply frontend/src/js/snack-message/SnackMessage.tsx]: I liked it that way. Took a look at 0.9, you are right, that looks better

**Review by Kadrian — APPROVED** [review]
Nice! Looks good to me

**Pull request merged**